File: Heap (Priority Queue)/3066. Minimum Operations to Exceed Threshold Value II.py
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        ans = 0
        arr = SortedList(nums)
        while len(arr) > 1 and k > arr[0]:
            ans += 1
            arr.add((arr[0] * 2) + arr[1])
            arr.discard(arr[1])
            arr.discard(arr[0])
        return ans

File: Heap (Priority Queue)/3066. Minimum Operations to Exceed Threshold Value II.cpp
class Solution {
    public:
        int minOperations(vector<int> &nums, int k)
        {
            priority_queue<long, vector<long>, greater<long>> minHeap;
            for (int &val : nums) minHeap.push(val);
    
            int ans = 0;
            while (minHeap.size() > 1 and k > minHeap.top())
            {            
                ans += 1;
                long minVal = minHeap.top(); minHeap.pop();
                long maxVal = minHeap.top(); minHeap.pop();
                minHeap.push(minVal * 2 + maxVal);
            }
            return ans;
        }
    };